### PR TITLE
get-trades implementation

### PR DESCRIPTION
**Pull request by Tiny-Angel305**

### CODE REVIEW

**Review by mustafapylons — APPROVED** [review]
Looks good to me.

File: lib/src/features/ipc/handlers/get_trades_handler.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        error: response.error,
+        errorCode: response.errorCode);
+    try {
+      if (response.success) {
+        defaultResponse.data = List.from(response.data).map((e) {
+          return Trade.create()..mergeFromProto3Json(e);
+        }).toList();
+      }
+    } on Exception catch (_) {
+      defaultResponse.error = 'Recipe parsing failed';
```
Comment (kjawadDeveloper2) [inline lib/src/features/ipc/handlers/get_trades_handler.dart:24]: Its not a recipe its a trade.

File: lib/src/pylons_wallet/pylons_wallet_impl.dart [hunk 2]
```diff
@@ -110,9 +110,13 @@ class PylonsWalletImpl implements PylonsWallet {
   }
 
   @override
-  Future<SDKIPCResponse> getTrades() async {
+  Future<SDKIPCResponse<List<Trade>>> getTrades() async {
     return Future.sync(() async {
-      return await _dispatch(Strings.GET_TRADES, '');
```
Comment (kjawadDeveloper2) [inline lib/src/pylons_wallet/pylons_wallet_impl.dart:115]: Get trades should be based on the creator.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]
Kindly change these things please.

File: lib/src/features/ipc/handlers/get_trades_handler.dart (hunk 1 above)
Comment (Tiny-Angel305) [reply lib/src/features/ipc/handlers/get_trades_handler.dart:24]: Thanks for pointing this out. It's fixed.

File: lib/src/pylons_wallet/pylons_wallet_impl.dart (hunk 2 above)
Comment (Tiny-Angel305) [reply lib/src/pylons_wallet/pylons_wallet_impl.dart:115]: @kjawadDeveloper2 , Can you review the changes please?

**Comment by kjawadDeveloper2** [comment]
The code is merged without running the test thats why trade test failed. @Tiny-Angel305  make sure it doesn't happen again. @mustafapylons i reverted the code.

**Pull request merged**